mesonbuild/wrap/wrap.py

```diff
@@ -525,8 +543,24 @@ def get_data(self, urlstring: str) -> T.Tuple[str, str]:
         elif WHITELIST_SUBDOMAIN in urlstring:
             raise WrapException(f'{urlstring} may be a WrapDB-impersonating URL')
         else:
+            headers = {}
+            is_http_protocol = url.scheme in ('http', 'https')
+            creds = self.get_netrc_credentials(url.netloc)
+
+            if is_http_protocol:
+                headers.update({'User-Agent': f'mesonbuild/{coredata.version}'})
+
+            req_urlstring = urlstring
+            if creds is not None and '@' not in url.netloc:
```

Comment:
What is the `@` check protecting here?

Reply:
Makes sense.

Reply:
I am not super familiar with what constitutes a valid URL. Can the `@` symbol appear in other sections of a URL unencoded?

Reply:
@tristan957 it checks only in the netloc part which is in the form `user:pass@example.com`. I don't think a domain name can include `@`.

Reply:
Missed the netloc part. Thanks!